### DISCUSSION TITLE
Re-scan Firebase collection for new fields

### DISCUSSION THREAD

**Discussion by tinrunje** [AntlerVC/firetable#1138] [Support / Q&A] [answered]
Is it possible to re-scan the collection for new fields? I have imported the fields upon creating the table, but I have created new firebase fields in the meantime. I tried adding them to Rowy one-by-one but they don't pull the data from Firebase then...

Is this possible?

**Comment by harinij** [comment, marked as answer]
Hi @tinrunje, If you have added a new field, then simply adding that field as a column by selectin the appropriate field type it should show up on the table. The only case in which it does not show up on the table is when an incompatible data type is picked. Alternatively, if you want to redo the setup of the table, you can have multiple tables connected to the same collection. You can think of Tables as "Views". So you can create another table and connect them to the same collection - which would help you with importing the data with the right field mapping suggestion. 

Hope this helps, if you still have issues, it would be help if you can share some screenshot of the data type of Firestore and Rowy - so that we can investigate this further.

**Reply by tinrunje** [reply]
Thanks! This "View" concept helps a lot, it makes it much clearer in the functionality and how it could be used. 
In terms of the field, I am talking about the **fcm_tokens** field as I believe I need to use it to send notifications.

Here is the image of the field:
<img width="979" alt="Screenshot 2023-02-27 at 21 57 58" src="https://user-images.githubusercontent.com/52575319/221684030-1319f2d8-275e-4adb-8fe7-0211a781ee97.png">

I tried adding it to Rowy with the data type being "JSON", but it doesn't add. But I also tried creating a new table from the same collection and the field still does not appear... Is this field special in some way?

**Reply by BeeBombshell** [reply]
Hey @tinrunje, as I see from this screenshot, the `fcm_token` is a sub-collection (collection inside a collection). This is the reason why it is not displaying appropriately when you import it as a JSON type. You'll have to add it as a **Sub-Table** column within your current table. Does this make sense? :)
